src/components/AddTransactionForm/AddTransactionForm.tsx

```diff
@@ -56,18 +71,23 @@ export const AddTransactionForm = ({
           currency: data.currency,
         },
         transactionFee: {
-          total: data.fee,
+          total: data.fee || 0,
           currency: data.currency,
         },
       };
 
-      const newTransactions = [...existingTransactions, transactionToSave];
-      localStorage.setItem('transactions', JSON.stringify(newTransactions));
+      if (transactionToEdit) {
+        editTransaction(transactionToSave);
+      } else {
+        addTransaction(transactionToSave);
+      }
 
       onClose();
 
       toast({
-        title: 'Transakce byla úspěšně přidána',
+        title: transactionToEdit
+          ? 'Transakce byla úspěšně upravena'
+          : 'Transakce byla úspěšně přidána',
         duration: 5000,
         className: 'bg-green-100 border-green-500 text-green-900',
```

Comment:
It would be good to create a utility function for toasts that will hold the common configuration, like class names for success and errors and duration.